sbc_compassion/models/sponsorship_correspondence.py

```diff
@@ -172,38 +171,50 @@ def get_communication_types(self):
             ('Supporter Letter', _('Supporter Letter')),
         ]
 
+    @api.multi
     @api.depends('sponsorship_id')
     def _set_name(self):
-        if self.sponsorship_id and self.communication_type_ids:
-            self.name = self.communication_type_ids[0].name + ' (' + \
-                self.sponsorship_id.partner_codega + " - " + \
-                self.child_id.code + ')'
-        else:
-            self.name = _('New correspondence')
+        for corr in self:
+            if corr.sponsorship_id and corr.communication_type_ids:
+                corr.name = corr.communication_type_ids[0].name + ' (' + \
+                            corr.sponsorship_id.partner_codega + " - " + \
+                            corr.child_id.code + ')'
+            else:
+                corr.name = _('New correspondence')
 
-    @api.depends('sponsorship_id', 'direction')
+    @api.depends('sponsorship_id', 'direction', 'original_language_id',
+                 'destination_language_id')
     def _set_languages(self):
```

Comment:
You can rename the method to `_set_destination_language` as we won't compute the original_language.